src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldPrefixFunction.java

```diff
@@ -18,32 +22,26 @@
  */
 class FieldPrefixFunction {
 
-    private final String delimiter;
+    private final CompositeStrings compositeStrings;
 
     FieldPrefixFunction(String delimiter) {
-        this.delimiter = delimiter;
+        Preconditions.checkArgument(delimiter.length() == 1, "only single character delimiters are supported");
```

Comment:
"single character" -> "single-character" (i.e., add hyphen) (also add in parallel test, of course)